stor/tests/test_dx.py

```diff
@@ -86,108 +88,212 @@ def test_resource_none_w_project(self):
             obj.write(b'hello')
+            obj.flush()
             obj.write(b' world')
-        self.assertIn(b'hello world', dx_p.read_object())
-
-    @mock.patch('time.sleep', autospec=True)
-    @mock.patch.object(DXPath, 'upload', autospec=True)
-    def test_write_multiple_flush_multiple_upload(self, mock_upload):
-        dx_p = DXPath('dx://project/obj')
-        with NamedTemporaryFile(delete=False) as ntf1,\
-                NamedTemporaryFile(delete=False) as ntf2,\
-                NamedTemporaryFile(delete=False) as ntf3:
-            with mock.patch('tempfile.NamedTemporaryFile', autospec=True) as ntf:
-                ntf.side_effect = [ntf1, ntf2, ntf3]
-                with dx_p.open(mode='wb') as obj:
-                    obj.write(b'hello')
-                    obj.flush()
-                    obj.write(b' world')
-                    obj.flush()
-                u1, u2, u3 = mock_upload.call_args_list
-                u1[0][1][0].source == ntf1.name
-                u2[0][1][0].source == ntf2.name
-                u3[0][1][0].source == ntf3.name
-                u1[0][1][0].object_name == dx_p.resource
-                u2[0][1][0].object_name == dx_p.resource
-                u3[0][1][0].object_name == dx_p.resource
-                self.assertEqual(open(ntf1.name).read(), 'hello')
-                self.assertEqual(open(ntf2.name).read(), 'hello world')
-                # third call happens because we don't care about checking for
-                # additional file change
-                self.assertEqual(open(ntf3.name).read(), 'hello world')
-
-
-@unittest.skip("skipping")
-class TestDXShared(SharedOBSFileCases, DXTestCase):
-    drive = 'dx://'
+            obj.flush()
+        f = dxpy.DXFile(dxid=dx_p.canonical_resource,
+                        project=dx_p.canonical_project)
+        # to allow for max of 20s for file state to go to closed
+        f.wait_on_close(20)
+        self.assertEqual(dx_p.open().read(), 'hello world')
+
+    def test_read_dir_fail(self):
+        self.setup_temporary_project()
+        self.setup_files(['/temp_folder/file'])
+        dx_p = DXPath('dx://' + self.project + ':/temp_folder')
+        with pytest.raises(exceptions.NotFoundError, match='No data object'):
+            dx_p.open().read()
+        dx_p = DXPath('dx://' + self.project)
+        with pytest.raises(ValueError, match='not a project'):
+            dx_p.open().read()
+
+    def test_write_to_project_fail(self):
+        self.setup_temporary_project()
+        dx_p = DXPath('dx://' + self.project)
+        with pytest.raises(ValueError, match='Cannot write to project'):
+            with dx_p.open(mode='wb') as f:
+                f.write(b'data')
+
+
+class TestDXOBSFile(SharedOBSFileCases, unittest.TestCase):
+    drive = 'dx://project:'  # project is required for DX paths
     path_class = DXPath
     normal_path = DXPath('dx://project:/obj')
 
+    def test_makedirs_p_does_nothing(self):
```

Comment:
`test_makedirs_p_does_nothing` is one of the tests inherited from `SharedOBSFileCases`, which works for other OBS, but won't for dx. `test_makedirs_p_does_something ` sounds good, but makedirs_p has a lot of different tests already in `class TestMakedirsP(DXTestCase)`.

No, since I am including the fixtures now, all of this should be in unit test. Also, `SharedOBSFileCases` are implemented as unit tests.